src/ngx_lua_resty_lmdb_module.c

```diff
@@ -254,6 +254,20 @@ static ngx_int_t ngx_lua_resty_lmdb_init_worker(ngx_cycle_t *cycle)
         return NGX_OK;
     }
 
+#if defined(HAVE_PRIVILEGED_PROCESS_PATCH) && !NGX_WIN32
+    if (ngx_process == NGX_PROCESS_HELPER && ngx_is_privileged_agent) {
+        int               i;
+        ngx_file_info_t  fi;
+
+        for (i = 0; i < 10; i++) {
```

Comment:
Should we log an error and return some non-zero value if the file/path does not exist after exhausting this loop? I can't imagine a scenario where things are _not_ broken if the file isn't created by another worker.

Also, if the agent worker creates the file it may require manual intervention from an operator to fix:

1) agent is started and waits
2) worker is started but crashes before creating `data.lmdb` (due to some unrelated/unknown reason)
3) agent finishes waiting and creates `data.lmdb`
4) `data.lmdb` now is owned by agent process uid/gid
5) nginx is unhealthy, so it is restarted
6) workers cannot read `data.lmdb`, so they all crash